pkg/activator/net/lb_policy.go

```diff
@@ -48,12 +48,12 @@ func randomChoice2(_ context.Context, targets []*podTracker) (func(), *podTracke
 		}, targets[0]
 	}
 	r1, r2 := 0, 1
-	// Two trackers - we know the both contestants,
+	// Two trackers - we know both contestants,
 	// otherwise pick 2 random unequal integers.
 	if l > 2 {
-		r1, r2 = rand.Intn(l), rand.Intn(l)
-		for r1 == r2 {
-			r2 = rand.Intn(l)
+		r1, r2 = rand.Intn(l), rand.Intn(l-1)
+		if r2 >= r1 {
+			r2++
```

Comment:
Do we need to check bounds? i.e `r2++ % l`

Reply:
no because second rand is l-1